### PR TITLE
fix: [PL-20756]: Edit in Token Form

### PR DESCRIPTION
**Pull request by reetika04**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by reetika04** [comment]
Screenshot 2022-03-14 at 12 09 41 PM

**Comment by reetika04** [comment]
retrigger jest

**Comment by reetika04** [comment]
retrigger rebase

File: src/modules/20-rbac/modals/TokenModal/views/TokenForm.tsx [hunk 1]
```diff
@@ -48,118 +47,124 @@ interface TokenFormData extends TokenDTO {
...
+      }}
+    >
+      {formikProps => {
+        return (
+          <Form>
+            <Container className={css.form}>
+              <ModalErrorHandler bind={setModalErrorHandler} />
+              <NameIdDescriptionTags
+                formikProps={formikProps}
+                identifierProps={{ isIdentifierEditable: !isEdit && !token }}
```
Comment (mayankVermaHarness) [inline src/modules/20-rbac/modals/TokenModal/views/TokenForm.tsx:137]: This is the main change in this PR, right?

**Review by mayankVermaHarness — APPROVED** [review]

File: src/modules/20-rbac/modals/TokenModal/views/TokenForm.tsx [hunk 2]
```diff
@@ -48,118 +47,124 @@ interface TokenFormData extends TokenDTO {
...
-          onSubmit?.()
-          onClose?.()
-        }
-      } else {
-        const created = await createToken(dataToSubmit)
-        /* istanbul ignore else */ if (created) {
-          showSuccess(getString('rbac.token.form.createSuccess', { name: values.name }))
-          onSubmit?.()
-          setToken(created.data)
+      switch (isEdit) {
```
Comment (avinashmadhwani02) [inline src/modules/20-rbac/modals/TokenModal/views/TokenForm.tsx:77]: why do we need a switch case for boolean values?

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/20-rbac/modals/TokenModal/views/TokenForm.tsx (hunk 1 above)
Comment (reetika04) [reply src/modules/20-rbac/modals/TokenModal/views/TokenForm.tsx:137]: Yes

File: src/modules/20-rbac/modals/TokenModal/views/TokenForm.tsx (hunk 2 above)
Comment (reetika04) [reply src/modules/20-rbac/modals/TokenModal/views/TokenForm.tsx:77]: Added it to reduce some cognitive complexity, which didn't work. So tried to minimize it elsewhere

**Pull request merged**